api/all/src/main/java/io/opentelemetry/api/common/ValueType.kt

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package io.opentelemetry.api.common
+
+/**
+ * AnyValue type options, mirroring [AnyValue#value
+ * options](https://github.com/open-telemetry/opentelemetry-proto/blob/ac3242b03157295e4ee9e616af53b81517b06559/opentelemetry/proto/common/v1/common.proto#L31).
+ *
+ * @since 1.42.0
+ */
+enum class ValueType {
```

Comment:
I wonder whether it's possible to represent values (and attributes more generally) in a sealed class hierarchy, as that could hide a lot of the current API surface. Ideally we'd avoid any type erasure with whatever approach we choose (if possible).

Reply:
defer until implementation?